source/common/stream_info/stream_info_impl.h

```diff
@@ -312,6 +314,16 @@ struct StreamInfoImpl : public StreamInfo {
     ASSERT(downstream_bytes_meter_.get() == downstream_bytes_meter.get());
   }
 
+  void setSanitizedPath(const Http::RequestHeaderMap& headers) override {
+    sanitized_path_ = Http::PathUtil::removeQueryAndFragment(headers.getPathValue());
+    auto pos = sanitized_path_.find_first_of(";");
+    if (pos != absl::string_view::npos) {
+      sanitized_path_.remove_suffix(sanitized_path_.length() - pos);
+    }
```

Comment:
Hi, why we need to remove suffix after `;` directly?

Reply:
If `ignore_port_in_host_matching` is set to true, the suffix after `;` will be removed. Like `"/path-bluh;env=prod"` will be sanitized to `"/path-bluh"`.

Reply:
OK, got it, thanks!

Reply:
> If `ignore_port_in_host_matching` in route configuration is set to true, the suffix after `;` will be removed. Like `"/path-bluh;env=prod"` will be sanitized to `"/path-bluh"`.

OK, got it. Thanks!